### PR TITLE
print descriptive state information string instead of integer

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Signed-off-by: Greg Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
@bboozzoo @pasinskim

**Comment by bboozzoo** [comment]
I'm quite sure it used to print state names instead of numbers. Was this changed recently?

**Comment by GregorioDiStefano** [comment]
Same here, I recall seeing the state change being a string, but we are back to showing only ID numbers..

**Comment by bboozzoo** [comment]
@pasinskim your doing?

**Comment by pasinskim** [comment]
I was also pretty sure we had the states printed, but I don't recall me changing this.

**Comment by pasinskim** [comment]
All right, now I know what happened here. For a while we've been using states as strings in real-time to get rid of the issue related to storing and loading state between updates. 
Later, I changed this to be int again and added custom marshaling before storing and reading: https://github.com/mendersoftware/mender/pull/169

This is the root-cause of the issue.

File: mender.go [hunk 1]
```diff
@@ -448,7 +448,7 @@ func (m mender) GetRetryPollInterval() time.Duration {
 }
 
 func (m *mender) SetState(s State) {
-	log.Infof("Mender state: %v -> %v", m.state.Id(), s.Id())
+	log.Infof("Mender state: %s -> %s", stateNames[m.state.Id()], stateNames[s.Id()])
```
Comment (pasinskim) [inline mender.go]: I would do this bit differently.

Just add `MenderState` `String()` method that will satisfy `Stringer` interface:
```
func (ms MenderState) String() string {
	return stateNames[ms]
}
```

Here you can have then:
`log.Infof("Mender state: %s -> %s", m.state.Id(), s.Id())`

Comment (GregorioDiStefano) [inline mender.go]: good suggestion, fixed

**Pull request merged**